building/qubes-r3-building.md

```diff
@@ -24,36 +38,80 @@ $ wget https://keys.qubes-os.org/keys/qubes-master-signing-key.asc
+Now let's bootstrap the builder. Unfortunately, the builder cannot verify itself (the classic Chicken and Egg problem), so we need to verify the signature manually:
 
 ~~~
 $ git clone git://github.com/QubesOS/qubes-builder.git
+$ mkdir qubes-builder/keyrings
+$ mkdir qubes-builder/keyrings/git
+$ cp .gnupg/pubring.gpg qubes-builder/keyrings/git/
+$ cp .gnupg/trustdb.gpg qubes-builder/keyrings/git/
 $ cd qubes-builder
-$ git describe --exact-match HEAD
-<some tag>
-$ git tag -v <some tag>
+$ git tag -v `git describe`
+~~~
+
+Assuming the verification went fine, we're good to go with all the rest without ever thinking more about verifying digital signatures on all the rest of the components.
+The builder will do that for us for each component, every time we build, even for all auxiliary files (e.g. Xen or Linux kernel sources).
+
+Let's configure the builder first (see Note at bottom if you would prefer to manually configure):
+
+~~~
+$ ./setup
+# Select 3.2 or 4.0 for version
+# Stable
+# No (we want a full build)
+# Select builder-fedora, builder-debian, template-whonix, mgmt-salt
+# Yes (to download)
+# Select fc26, stretch, whonix-gateway, whonix-workstation (for the currently shipping templates)
 ~~~
 
-Assuming the verification went fine, we're good to go with all the rest without ever thinking more about verifying digital signatures on all the rest of the components, as the builder will do that for us, for each component, every time we, even for all aux files (e.g. Xen or Linux kernel sources).
+Once the download is complete, continue the build process with:
+
+~~~
+$ make install-deps
+$ make get-sources
+~~~
 
-Let's configure the builder first (we can use one of the example configs, either for R2 or "master", which currently means pre-released R3):
+Finally, if you are making a test build, use:
 
 ~~~
-cp example-configs/qubes-os-master.conf builder.conf
+$ make qubes
+$ make iso
 ~~~
 
-You can take a loot at the `builder.conf.default` for a description of all available options. Nevertheless, the default config should be enough for start:
+Or for a fully signed build (this requires setting SIGN_KEY in the builder.conf):
 
 ~~~
-$ make get-sources qubes
-$ make sign-all # this requires setting SIGN_KEY in the builder.conf, can be skipped for test builds.
+$ make sign-all
 $ make iso
 ~~~
 
 Enjoy your new ISO!
+
+
+**Note** Instead of using `./setup`, you can manually configure the build by doing `cp example-configs/qubes-os-master.conf builder.conf` and editing `builder.conf`.
+Take a look at `builder.conf.default` for a description of all available options.
+On manually configured builds, you may also need to:
+
+~~~
+export GNUPGHOME=~/qubes-builder/keyrings/git
+mkdir --parents "$GNUPGHOME"
+chmod --recursive 700 "$GNUPGHOME"
+echo '427F11FD0FAA4B080123F01CDDFA1A3E36879494:6:' | gpg --import-ownertrust
+~~~
```

Comment:
This will be done automatically on first `make get-sources` call. Unless you create the directory manually but forget to import keys there, which is exactly what you've done here...

Reply:
Don't these accomplish the same thing?
```
$ cp .gnupg/pubring.gpg qubes-builder/keyrings/git/
$ cp .gnupg/trustdb.gpg qubes-builder/keyrings/git/
```
But I'll remove.

Reply:
Yes, indeed. But generally IMO better to simplify the instruction as much as possible. The critical stuff is to validate if downloaded qubes-builder is properly signed, that should be the last thing one need to do with gpg manually. All further steps should be handled by scripts in qubes-builder. Less space for human error.